manim/mobject/types/vectorized_mobject.py

```diff
@@ -920,10 +982,23 @@ def point_from_proportion(self, alpha: float) -> np.ndarray:
         np.ndarray
             Point evaluated.
         """
-        num_cubics = self.get_num_curves()
-        n, residue = integer_interpolate(0, num_cubics, alpha)
-        curve = self.get_nth_curve_function(n)
-        return curve(residue)
+
+        curves_with_lengths = list(self.get_curve_functions_with_lengths())
+
+        total_length = np.sum(length for _, length in curves_with_lengths)
+        target_length = alpha * total_length
+        current_length = 0
+
+        for curve, length in curves_with_lengths:
+            if current_length + length >= target_length:
+                if length != 0:
```

Comment:
```suggestion
                if length:
```

Reply:
Idk, I'm really not a fan of implicit bool conversions in most cases. For implicit bool conversions you're shown the variable, then you have to realize that it's not a bool even though it's being treated as one, then you have to think about what type the variable is, then you have to remember in which way that type converts to a bool. With more explicit things however, like `length != 0`, then it's clear and direct: it evaluates to true when `length` doesn't equal 0, which also makes it clearer that the next line would break if `length` were to equal 0; it's more readable. And as the zen of python tells us, explicit is better than implicit and readability counts.

I know implicit bool conversions are used elsewhere in the library though, so if the consensus is that it should just be `if length:` then that's fine

Reply:
Personally, I'd use `if length > 0` here -- but I don't think this PR should be blocked/stalled over this.

Would you be okay with leaving it as-is, @jsonvillanueva?

Reply:
Yeah I'd be cool with `length > 0` too, but in my mind, since it's not possible for `length` to be negative (as it's a sum of vector magnitudes), I only needed to handle the case of `0` which is also the most flagrantly broken case as `length` is used as the divisor. That and it'd be kinda weird to use `residue = 0` for negative length as a negative length seems like it would be a deeper issue

Reply:
Sure, we can leave it as is